### PR TITLE
Add Exercise01

### PR DESCRIPTION
**Pull request by huynhngocminh2511**

### CODE REVIEW

**Review by voquanghoa — COMMENTED** [review]
Please remove unnecessary files

File: Exercise01/CustomRecyclerView/CustomAdapter.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+using System.Collections.Generic;
+using Exercise01.CustomRecyclerView.Models;
+using Android.Util;
+using Android.Graphics;
+
+namespace Exercise01.CustomRecyclerView
+{
+    class CustomAdapter : RecyclerView.Adapter
+    {
+        List<Item> items;
```
Comment (voquanghoa) [inline Exercise01/CustomRecyclerView/CustomAdapter.cs]: Make it static and rename to `androidVersions`.

File: Exercise01/CustomRecyclerView/CustomAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+        public CustomAdapter()
```
Comment (voquanghoa) [inline Exercise01/CustomRecyclerView/CustomAdapter.cs]: Remove unused contructor

File: Exercise01/CustomRecyclerView/Models/Item.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+using Android.App;
+using Android.Content;
+using Android.OS;
+using Android.Runtime;
+using Android.Views;
+using Android.Widget;
+
+namespace Exercise01.CustomRecyclerView.Models
+{
+    class Item
```
Comment (voquanghoa) [inline Exercise01/CustomRecyclerView/Models/Item.cs]: Class name should be changed to `AndroidVersion`

File: Exercise01/MainActivity.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+            using (var streamReader = new StreamReader(input))
+            {
+                var content = streamReader.ReadToEnd();
+                var items = JsonConvert.DeserializeObject<List<Item>>(content);
+
+                adapter = new CustomAdapter(items);
+            }
+
+            recyclerView.SetAdapter(adapter);
```
Comment (voquanghoa) [inline Exercise01/MainActivity.cs]: Move this `recyclerView.SetAdapter(adapter);` to the `using` scope and make `adapter` is inner variable

File: Exercise01/CustomRecyclerView/CustomAdapter.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        {
+            var item = items[position];
+
+            // Replace the contents of the view with that element
+            var holder = viewHolder as CustomAdapterViewHolder;
+
+            var imageAsBytes = Base64.Decode(item.Image, Base64Flags.Default);
+            var imageAsBitmap = BitmapFactory.DecodeByteArray(imageAsBytes, 0, imageAsBytes.Length);
+
+            holder.ImageViewIcon.SetImageBitmap(imageAsBitmap);
```
Comment (voquanghoa) [inline Exercise01/CustomRecyclerView/CustomAdapter.cs]: You should move this logic to the class `CustomAdapterViewHolder`

File: Exercise01/CustomRecyclerView/CustomAdapter.cs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            holder.TextViewName.Text = item.Name;
+            holder.TextViewVersion.Text = item.Version;
+        }
+
+        public override int ItemCount => items.Count;
+    }
+
+    public class CustomAdapterViewHolder : RecyclerView.ViewHolder
+    {
+        public ImageView ImageViewIcon { get; set; }
```
Comment (voquanghoa) [inline Exercise01/CustomRecyclerView/CustomAdapter.cs]: Let this properties to become private fields

File: Exercise01/MainActivity.cs (hunk 4 above)
Comment (huynhngocminh2511) [reply Exercise01/MainActivity.cs]: Done

File: Exercise01/CustomRecyclerView/Models/Item.cs (hunk 3 above)
Comment (huynhngocminh2511) [reply Exercise01/CustomRecyclerView/Models/Item.cs]: Done

File: Exercise01/CustomRecyclerView/CustomAdapter.cs (hunk 6 above)
Comment (huynhngocminh2511) [reply Exercise01/CustomRecyclerView/CustomAdapter.cs]: Done

File: Exercise01/CustomRecyclerView/CustomAdapter.cs (hunk 5 above)
Comment (huynhngocminh2511) [reply Exercise01/CustomRecyclerView/CustomAdapter.cs]: Done

File: Exercise01/CustomRecyclerView/CustomAdapter.cs (hunk 2 above)
Comment (huynhngocminh2511) [reply Exercise01/CustomRecyclerView/CustomAdapter.cs]: Done

File: Exercise01/CustomRecyclerView/CustomAdapter.cs (hunk 1 above)
Comment (huynhngocminh2511) [reply Exercise01/CustomRecyclerView/CustomAdapter.cs]: Your mind is make it private?

**Pull request merged**